scripts/pylint_extensions.py

```diff
@@ -1252,12 +1253,15 @@ def process_module(self, node):
             # shebang expression at the start of a bash script which
             # loses function when a space is added.
             if re.search(br'^#[^\s].*$', line) and not line.startswith(b'#!'):
+                space_at_beginning = False
                 self.add_message(
                     'no-space-at-beginning', line=line_num + 1)
 
-            # Check if comment contains version info or data type.
-            if not previous_line.startswith(b'#'):
-                data_type_is_present = any(word in line for word in DATA_TYPES)
+            # Check if comment contains version info or data type
+            # at the beginning.
+            if line.startswith(b'#') and len(line) > 1 and space_at_beginning:
+                word = line.split()[1]
+                data_type_is_present = word in DATA_TYPES
                 if data_type_is_present or re.search(br'^# v[0-9]+ .*$', line):
```

Comment:
Can you explain what the "version info" bit means here? I don't really understand it.

I also feel like something is a bit "off" about this whole method. I would have expected something along the lines of "extract the docstring", "extract the args/returns section", "make sure each arg is correctly formatted". The current implementation seems to be a bit of a line-by-line check, which seems error-prone because context isn't really being maintained about "where we are in the file" -- so it all seems a bit ad hoc. In addition I know there are other typeinfo present besides the ones in DATA_TYPES so this check is probably incomplete.

So, in general, I am having trouble understanding what is going on with this code. Could we talk about it? I'll set up a meeting for this.

Reply:
Per discussion with @seanlip found that he was misinterpreting this as a PR for docstrings.

Reply:
As per discussion, I have added more documentation in the code.

Reply:
Per discussion with @seanlip found that he was misinterpreting this as a PR for docstrings. I have added more documentation in the code.